### DISCUSSION TITLE
Enable support like this Placeholders for  X-Forwarded-Proto

### DISCUSSION THREAD

**Discussion by rocleegithub** [ThreeMammals/Ocelot#2195] [Ideas]
##  New Feature

Enable support like this Placeholders

`"X-Forwarded-Proto": "{UpstreamForwardedProto}"`

Enable support for this for some services

Thanks~

**Comment by raman-m** [comment]
Didn't you read **[Contributing](https://github.com/ThreeMammals/Ocelot?tab=readme-ov-file#contributing)** rules in [ReadMe.md](https://github.com/ThreeMammals/Ocelot/blob/develop/README.md#contributing)?
You need to use [Issue template](https://github.com/ThreeMammals/Ocelot/blob/main/.github/ISSUE_TEMPLATE.md) to write the description!

> Enable support like this Placeholders
> `"X-Forwarded-Proto": "{UpstreamForwardedProto}"`

Could you provide more details of your idea please, and what's the final goal of your request?
What Ocelot feature is it related to?

> Enable support for this for some services

What' services did you mean actually?

**Comment by rocleegithub** [comment]
Restore the original Request Scheme through `Request.GetAbsoluteUri() `

**Reply by raman-m** [reply]
Sorry, I don't understand you!

**Comment by rocleegithub** [comment]
like this  `proxy_set_header X-Forwarded-Proto $http_X_Forwarded_Proto??$scheme;`

**Reply by raman-m** [reply]
We are discussing [Headers Transformation](https://ocelot.readthedocs.io/en/latest/features/headerstransformation.html) and, in particular, [Placeholders](https://ocelot.readthedocs.io/en/latest/features/headerstransformation.html#placeholders), correct? The currently implemented [placeholders](https://ocelot.readthedocs.io/en/latest/features/headerstransformation.html#placeholders) for forwarding include:
- [{RemoteIpAddress}](https://ocelot.readthedocs.io/en/latest/features/headerstransformation.html#x-forwarded-for), which corresponds to `HttpContext.Connection.RemoteIpAddress`
- `{UpstreamHost}`, which is equivalent to `HttpContext.Request.Headers.TryGetValue("Host", out var upstreamHost)`

All placeholders versus headers are implemented in the `Ocelot.Infrastructure.Placeholders` class &rarr;
https://github.com/ThreeMammals/Ocelot/blob/da9d6fab6c704baa925d3c1c5fc984f6a9e9c717/src/Ocelot/Infrastructure/Placeholders.cs#L9

----
> "X-Forwarded-Proto": "{UpstreamForwardedProto}"

I've reviewed the Mozilla documentation and realized that the [X-Forwarded-Proto](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto) header is [not a standard](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto#specifications) one; internet tools use their own headers (refer to [Examples](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Forwarded-Proto#examples)). Therefore, we have a standardized header, which is [Forwarded](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded). It appears that you do not have this [Forwarded](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded) header in downstreams, correct?

#### In conclusion,
it would be logical to include support for those headers; at the very least, we could treat the `X-Forwarded-Proto` header as a proprietary Ocelot header.
**Would you be willing to initiate a pull request?**

**Reply by raman-m** [reply]
Current PR is #2201
- #2201
